Dfe.Academies.Academisation.Data.UnitTest/ProjectAggregate/ProjectUpdateDataCommandTests.cs

```diff
@@ -0,0 +1,97 @@
+﻿using System.Threading.Tasks;
+using AutoFixture;
+using Dfe.Academies.Academisation.Data.ProjectAggregate;
+using Dfe.Academies.Academisation.Data.UnitTest.Contexts;
+using Dfe.Academies.Academisation.Domain.Core.ProjectAggregate;
+using Dfe.Academies.Academisation.Domain.ProjectAggregate;
+using Dfe.Academies.Academisation.IDomain.ProjectAggregate;
+using Microsoft.EntityFrameworkCore;
+using Xunit;
+
+namespace Dfe.Academies.Academisation.Data.UnitTest.ProjectAggregate;
+
+public class ProjectUpdateDataCommandTests
+{
+	private readonly Fixture _fixture = new();
+
+	private readonly ProjectUpdateDataCommand _subject;
+	private readonly AcademisationContext _context;
+
+	public ProjectUpdateDataCommandTests()
+	{
+		_context = new TestProjectContext().CreateContext();
+		_subject = new ProjectUpdateDataCommand(_context);
+	}
+
+	[Fact]
+	public async Task ProjectExists___ProjectUpdated()
+	{
+		// arrange
+		ProjectDetails projectDetails = _fixture.Create<ProjectDetails>();		
+
+		IProject project = new Project(1, projectDetails);
+		await _context.Projects.LoadAsync();
+
+		// act
+		await _subject.Execute(project);
+
+		_context.ChangeTracker.Clear();
+
+		var updatedProject = await _context.Projects.SingleAsync(p => p.Id == project.Id);
+
+		// assert
+		AssertProjectModelIsEqual(project.Details, updatedProject);
+	}
+
+	private static void AssertProjectModelIsEqual(ProjectDetails expectedProject, ProjectState actualProject)
+	{
+		Assert.Multiple(
```

Comment:
Could replace all these fieldwise asserts with an Assert.Equivalent, I think?

Reply:
I'm pretty sure, I tried that. I can't remember why it didn't work. Will try again and let you know.

Reply:
Yeah, it doesn't work because the objects are slightly different.
Should they be the same?
 Assert.Equivalent() Failure: Mismatched member list
    Expected: ["EqualityContract", "Urn", "Laestab", "SchoolName", "LocalAuthority", "ApplicationReferenceNumber", "UkPrn", "ProjectStatus", "ApplicationReceivedDate", "AssignedDate", "HeadTeacherBoardDate", "OpeningDate", "BaselineDate", "LocalAuthorityInformationTemplateSentDate", "LocalAuthorityInformationTemplateReturnedDate", "LocalAuthorityInformationTemplateComments", "LocalAuthorityInformationTemplateLink", "LocalAuthorityInformationTemplateSectionComplete", "RecommendationForProject", "Author", "Version", "ClearedBy", "AcademyOrderRequired", "PreviousHeadTeacherBoardDateQuestion", "PreviousHeadTeacherBoardDate", "PreviousHeadTeacherBoardLink", "TrustReferenceNumber", "NameOfTrust", "SponsorReferenceNumber", "SponsorName", "AcademyTypeAndRoute", "ProposedAcademyOpeningDate", "SchoolAndTrustInformationSectionComplete", "ConversionSupportGrantAmount", "ConversionSupportGrantChangeReason", "PublishedAdmissionNumber", "PartOfPfiScheme", "ViabilityIssues", "FinancialDeficit", "DistanceFromSchoolToTrustHeadquarters", "DistanceFromSchoolToTrustHeadquartersAdditionalInformation", "MemberOfParliamentParty", "MemberOfParliamentName", "GeneralInformationSectionComplete", "SchoolPerformanceAdditionalInformation", "RationaleForProject", "RationaleForTrust", "RationaleSectionComplete", "RisksAndIssues", "EqualitiesImpactAssessmentConsidered", "RisksAndIssuesSectionComplete", "RevenueCarryForwardAtEndMarchCurrentYear", "ProjectedRevenueBalanceAtEndMarchNextYear", "CapitalCarryForwardAtEndMarchCurrentYear", "CapitalCarryForwardAtEndMarchNextYear", "SchoolBudgetInformationAdditionalInformation", "SchoolBudgetInformationSectionComplete", "CurrentYearCapacity", "CurrentYearPupilNumbers", "YearOneProjectedCapacity", "YearOneProjectedPupilNumbers", "YearTwoProjectedCapacity", "YearTwoProjectedPupilNumbers", "YearThreeProjectedCapacity", "YearThreeProjectedPupilNumbers", "SchoolPupilForecastsAdditionalInformation", "KeyStage2PerformanceAdditionalInformation", "KeyStage4PerformanceAdditionalInformation", "KeyStage5PerformanceAdditionalInformation", "Upin", "NewAcademyUrn"]
    Actual:   ["Urn", "Laestab", "SchoolName", "LocalAuthority", "ApplicationReferenceNumber", "UkPrn", "ProjectStatus", "ApplicationReceivedDate", "AssignedDate", "HeadTeacherBoardDate", "OpeningDate", "BaselineDate", "LocalAuthorityInformationTemplateSentDate", "LocalAuthorityInformationTemplateReturnedDate", "LocalAuthorityInformationTemplateComments", "LocalAuthorityInformationTemplateLink", "LocalAuthorityInformationTemplateSectionComplete", "RecommendationForProject", "Author", "Version", "ClearedBy", "AcademyOrderRequired", "PreviousHeadTeacherBoardDateQuestion", "PreviousHeadTeacherBoardDate", "PreviousHeadTeacherBoardLink", "TrustReferenceNumber", "NameOfTrust", "SponsorReferenceNumber", "SponsorName", "AcademyTypeAndRoute", "ProposedAcademyOpeningDate", "SchoolAndTrustInformationSectionComplete", "ConversionSupportGrantAmount", "ConversionSupportGrantChangeReason", "PublishedAdmissionNumber", "PartOfPfiScheme", "ViabilityIssues", "FinancialDeficit", "DistanceFromSchoolToTrustHeadquarters", "DistanceFromSchoolToTrustHeadquartersAdditionalInformation", "MemberOfParliamentParty", "MemberOfParliamentName", "GeneralInformationSectionComplete", "SchoolPerformanceAdditionalInformation", "RationaleForProject", "RationaleForTrust", "RationaleSectionComplete", "RisksAndIssues", "EqualitiesImpactAssessmentConsidered", "RisksAndIssuesSectionComplete", "RevenueCarryForwardAtEndMarchCurrentYear", "ProjectedRevenueBalanceAtEndMarchNextYear", "CapitalCarryForwardAtEndMarchCurrentYear", "CapitalCarryForwardAtEndMarchNextYear", "SchoolBudgetInformationAdditionalInformation", "SchoolBudgetInformationSectionComplete", "CurrentYearCapacity", "CurrentYearPupilNumbers", "YearOneProjectedCapacity", "YearOneProjectedPupilNumbers", "YearTwoProjectedCapacity", "YearTwoProjectedPupilNumbers", "YearThreeProjectedCapacity", "YearThreeProjectedPupilNumbers", "SchoolPupilForecastsAdditionalInformation", "KeyStage2PerformanceAdditionalInformation", "KeyStage4PerformanceAdditionalInformation", "KeyStage5PerformanceAdditionalInformation", "Upin", "NewAcademyUrn", "Id", "CreatedOn", "LastModifiedOn"]